test/wrappers/vroom_test.rb

```diff
@@ -791,4 +791,26 @@ def test_quantity_precision
       assert_equal 1.001, activity[:detail][:quantities].first[:value]
     }
   end
+
+  def test_partially_nil_capacities
+    problem = VRP.basic
+    problem[:services].each{ |service|
+      service[:quantities] = [{ unit_id: 'kg', value: 1 }]
+    }
+    problem[:vehicles] << problem[:vehicles].first.dup
+    problem[:vehicles].first[:capacities] = [{ unit_id: 'kg', limit: 2 }]
+    problem[:vehicles].last[:id] = 'vehicle_1'
+
+    vroom = Wrappers::Vroom.new
+    vroom.stub(
+      :run_vroom, lambda{ |vroom_vrp, _job|
+        assert_equal [2 * Wrappers::Vroom::CUSTOM_QUANTITY_BIGNUM], vroom_vrp[:vehicles].first[:capacity]
+        assert_equal vroom_vrp[:jobs].flat_map{ |job| job[:pickup].first }.sum,
+                     vroom_vrp[:vehicles].last[:capacity].first
+        nil
+      }
+    ) do
+      vroom.solve(TestHelper.create(problem))
+    end
```

Comment:
I am not sure about second assert. What are we checking ? I would say we need to check result.

```suggestion
   vroom_vrp = vroom.vroom_problem(vrp, [:time, :distance])
   expected_capacity = problem[:vehicles].first[:capacitities].first[:limit]
   assert_equal [expected_capacity * Wrappers::Vroom::CUSTOM_QUANTITY_BIGNUM], vroom_vrp[:vehicles].first[:capacity]
   
   result = vroom.run_vroom(vroom_vrp, nil)
   assert_operator result['routes'].first.sum{ |a| ['service', 'pickup', 'delivery'].include?('a[:type]) ? a[:detail][:quantities].first[:value] : 0 }, :<=, problem[:vehicles].first[:capacity].first[:limit]
```

Reply:
We are checking that the capacity of the vehicle with a capacity set to nil is equal to the sum of all the demands

Reply:
Vroom is not called here

Reply:
Oh okay I read 'first' instead of 'last', sorry :)